packages/cmrjs/src/cmr-utils.js

```diff
@@ -44,6 +44,27 @@ function getS3UrlOfFile(file) {
   throw new Error(`Unable to determine location of file: ${JSON.stringify(file)}`);
 }
 
+function getFilename(file) {
+  if (file.fileName) return file.fileName;
+  if (file.name) return file.name;
+  if (file.filename) return path.basename(file.filename);
+  if (file.filepath) return path.basename(file.filepath);
+  if (file.key) return path.basename(file.key);
+  throw new Error(`Unable to determine name of file: ${JSON.stringify(file)}`);
+}
+
+function getFileDescription(file) {
```

Comment:
If `getFilename` returns `undefined` in the failure case then this can just be:

```js
  const filename = getFilename(file);

  return filename ? `Download ${filename}` : 'File to download';
```